pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1393,7 +1393,37 @@ def _check_unnecessary_comprehension(self, node):
         else:
             return
         if expr_list == target_list != []:
-            self.add_message("unnecessary-comprehension", node=node)
+            args = None
+            if isinstance(node.parent, astroid.DictComp) and isinstance(
+                utils.safe_infer(node.iter), astroid.objects.DictItems
+            ):
+                args = (f"{node.iter.func.expr.as_string()}",)
+            if (
```

Comment:
Can this be `elif`?

Reply:
We have a checker for that, we should add some optional checker in our pylintrc :smile:

Reply:
elif seems to work

Reply:
Fixed: 0912d67